src/main/java/com/palantir/gradle/graal/BaseGraalCompileTask.java

```diff
@@ -96,18 +104,206 @@ protected final String generateClasspathArgument() {
         classpathArgument.addAll(classpath.get().getFiles());
         classpathArgument.add(jarFile.getAsFile().get());
 
-        return classpathArgument.stream().map(File::getAbsolutePath).collect(Collectors.joining(":"));
+        return classpathArgument.stream()
+            .map(File::getAbsolutePath)
+            .collect(Collectors.joining(getArchitectureSpecifiedPathSeparator()));
     }
 
     private Path getArchitectureSpecifiedBinaryPath() {
         switch (Platform.operatingSystem()) {
             case MAC: return Paths.get("Contents", "Home", "bin", "native-image");
             case LINUX: return Paths.get("bin", "native-image");
+            case WINDOWS: return Paths.get("bin", "native-image.cmd");
+            default:
+                throw new IllegalStateException("No GraalVM support for " + Platform.operatingSystem());
+        }
+    }
+
+    private String getArchitectureSpecifiedPathSeparator() {
+        switch (Platform.operatingSystem()) {
+            case MAC:
+            case LINUX:
+                return ":";
+            case WINDOWS:
+                return ";";
             default:
                 throw new IllegalStateException("No GraalVM support for " + Platform.operatingSystem());
         }
     }
 
+    protected final void configurePlatformSpecifics(ExecSpec spec) {
+        if (Platform.operatingSystem() == Platform.OperatingSystem.WINDOWS) {
+            // on Windows the native-image executable needs to be launched from the Windows SDK Command Prompt
+            // this is mentioned at https://github.com/oracle/graal/tree/master/substratevm#quick-start
+            // all this does though, is setting a bunch of environment variables
+            // the main ones are set here to avoid the need for running inside the Windows SDK Command Prompt
+            // reference (with installed SDK): C:\Program Files\Microsoft SDKs\Windows\v7.1\Bin\SetEnv.cmd
+            // assuming x64 as current and target cpu
```

Comment:
The alternative is to call `SetEnv.cmd` from the Microsoft SDK. See https://github.com/palantir/gradle-graal/pull/139/files#diff-31212796681fcca0c36a8b735d54c700R52. Since GraalVM depends on that SDK, I see no differences. In case the SDK will get udpated in the context of GraalVM, both solutions (call the `cmd`) and the one here, need to be adapted.

The solution by @frozenice reads more self-contained (as no separate `cmd` file needs to be created during building).

Reply:
The alternative is to call `SetEnv.cmd` from the Microsoft SDK. See https://github.com/palantir/gradle-graal/pull/139/files#diff-31212796681fcca0c36a8b735d54c700R52. Since GraalVM depends on that SDK, I see no differences. In case the SDK will get udpated in the context of GraalVM, both solutions (call the `cmd`) and the one here, need to be adapted.

The solution by @frozenice reads more self-contained (as no separate `cmd` file needs to be created during building).

Reply:
The registry reads are just what the `SetEnv.cmd` does. @koppor's solution is nice, too. I hadn't consired this method, because I had already formulated the registry plan in my mind...

I might be in favor of his method, because it's easier to adapt, if Graal supports other SDK versions in the future (and is generally less code). I don't think there are privilege issues with the registry access, but this would be a non-issue with his method (of course, the `SetEnv.cmd` also calls reg.exe, but we don't care what it does exactly, which I find nice).

@koppor do you want to do a PR against [my forked feature-branch](https://github.com/frozenice/gradle-graal/tree/windows) with your method, replacing my registry stuff? You could replace `Files.createTempDirectory("com.palantir")` with `getProject().getBuildDir().toPath().resolve("tmp").resolve("com.palantir.graal")` or something like this, so the temporary .cmd file is contained in `$buildDir` and can be `:clean`ed.

Btw, the full command run by the *Windows SDK 7.1 Command Prompt* shortcut is:

```
C:\Windows\System32\cmd.exe /E:ON /V:ON /T:0E /K "C:\Program Files\Microsoft SDKs\Windows\v7.1\Bin\SetEnv.cmd"
```
You might want to add `/E:ON` (command extensions) and `/V:ON` (delayed environment variable expansion via !) to your cmd.exe call, too.